src/components/Menu/MenuItem/MenuItem.tsx

```diff
@@ -290,16 +290,17 @@ const MenuItem: VibeComponent<MenuItemProps> & {
         onMouseEnter={onMouseEnter}
         tabIndex={TAB_INDEX_FOCUS_WITH_JS_ONLY}
       >
-        {renderMenuItemIconIfNeeded()}
-
         <Tooltip
           content={shouldShowTooltip ? finalTooltipContent : null}
           position={tooltipPosition}
           showDelay={tooltipShowDelay}
         >
-          <div ref={titleRef} className="monday-style-menu-item__title">
-            {title}
-          </div>
+          {renderMenuItemIconIfNeeded()}
+          {title && (
+            <div ref={titleRef} className="monday-style-menu-item__title">
+              {title}
+            </div>
+          )}
         </Tooltip>
         {label && (
           <div ref={titleRef} className="monday-style-menu-item__label">
```

Comment:
Is it ok that we might have 2 titleRefs? Is there a difference between `title` and `label` or it's just need for backward compatibility?